ui/app/src/components/DeleteProjectDialog/DeleteProjectDialog.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DispatchWithoutAction } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, IconButton } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface DeleteProjectDialogProps {
+  name: string | undefined;
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: DispatchWithoutAction;
+}
+
+/**
+ * Dialog used to delete a project.
+ *
+ * @param props.name The name of the project to delete.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @constructor
+ */
+const DeleteProjectDialog = (props: DeleteProjectDialogProps) => {
+  const { name, open, onClose, onConfirm } = props;
+
+  const triggerOnConfirm = () => {
+    onClose();
+    onConfirm();
+  };
+  const triggerOnClose = () => {
+    onClose();
+  };
+  return (
+    <Dialog open={open} onClose={triggerOnClose}>
+      <DialogTitle>Delete Project</DialogTitle>
+      <IconButton
+        aria-label="Close"
+        onClick={() => onClose()}
+        sx={(theme) => ({
+          position: 'absolute',
+          top: theme.spacing(0.5),
+          right: theme.spacing(0.5),
+        })}
+      >
+        <CloseIcon />
+      </IconButton>
+      <DialogContent dividers sx={{ width: '500px' }}>
+        Are you sure you want to delete Project {name}? This will delete all the dashboards within the project.
```

Comment:
I notice we sometimes use capitalized `Project` and sometimes use lowercase `project`... maybe we want to consistently use capitalized or lowercase? 

I would vote for lowercase `project`

Reply:
By the way, I think this explanation is really good ("This will delete all the dashboards within the project")

Reply:
> By the way, I think this explanation is really good ("This will delete all the dashboards within the project")

A shameful copy paste from @eunicorn's code on panel group. All the credits go to her :)
![image](https://user-images.githubusercontent.com/10258608/214536048-8a44f735-bc25-416d-a848-eee6e4bb7ece.png)

> I notice we sometimes use capitalized `Project` and sometimes use lowercase `project`... maybe we want to consistently use capitalized or lowercase?
> 
> I would vote for lowercase `project`

Yes this is something haunting us as well in another UI we maintain haha. We ended up using a displayKind() utils function but not always used... 

The rule of thumb that I can take from your comments would be : 
- in a sentence use ``project``
- in a title or button use ``Project``
I will go for it.